internal/provider_test/integration_test.go

```diff
@@ -1337,3 +1337,87 @@ func checkDataSourceList(t *testing.T, dataSourceType, dataSource string, expect
 		tf.CheckStateResource(t, address).ForEach(dataSourceType, expectedCount, assertFn)
 	})
 }
+
+func TestValidation(t *testing.T) {
+	// Create provider server that runs within test
+	ctx := context.Background()
+	reattachCfg, closeFn := CreateProviderServer(t, ctx)
+	defer closeFn()
+
+	// Create Terraform workspace and initialize it with config
+	tf := CreateTerraformWorkspace(t)
+	err := tf.SetReattachConfig(reattachCfg)
+	require.NoError(t, err)
+
+	t.Run("invalid project name", func(t *testing.T) {
+		vars := newTestVars(false)
+		projName := "this is not a valid project name"
+
+		vars.project.Name = projName
+
+		tf.WriteConfigT(t, vars.builder.Build())
+		_, err = tf.Init()
+		require.NoError(t, err)
```

Comment:
Why don't you just move this out? I don't think any of the subsequent `terraform init` invocations do anything since the same workspace is reused.

Reply:
We need to write the config at least once before calling `init`, so that terraform knows what providers need to be initialized.

On further experimentation, because we are only using our provider and that provider is already initialized via the reattach config, we don't need to `init` at all.

Reply:
Right, when I said "move this out", I meant to the outer test function before all of these subtests are run.

I make sure to invoke `terraform init` once in each test function, so I'm not sure about omitting it entirely.

Reply:
If we want to move the init outside of the sub-tests, we also need to write the (default?) config before we run it.

In any case, it is a moot point because the test runs fine without init.

Reply:
Correct. All the other init invocations are made after the configuration is written for the first time.